transmitter.js

```diff
@@ -1,28 +1,15 @@
-let transmissionFailureCount = 0;
-
-function networkTransmitStub(celsius) {
-    console.log(`Temperature to transmit: ${celsius} Celsius`);
-    // Generate a random number between 0 and 1
-    const randomNumber = Math.random();
-    // Map the random number to either 200 or 500
-    // If randomNumber is less than 0.5, return 200, otherwise return 500
-    return randomNumber < 0.5 ? 200 : 500;
-
-    // The intention here is to transmit the temperature over a network.
-    // However, this is a stub, so we just print the message above
-    // and give back a return code.
-
+let transmissionFailureCount=0;
+function transmitInCelcius(fahrenheit, networkTransmit = networkTransmitStub) {
+  const celsius = (fahrenheit - 32) * 5 / 9;
+  const returnCode = networkTransmit(celsius);
+  if (returnCode !== 200) {
+    // Increment failure count for non-200 return codes
+    transmissionFailureCount ++;
+  }
 }
-
-function transmitInCelcius(fahrenheit) {
-    const celsius = (fahrenheit - 32) * 5 / 9;
-    const returnCode = networkTransmitStub(celsius);
-    if (returnCode !== 200) {
-        // non-ok response indicates failure while transmitting over the network
-        // record of failure count
-        transmissionFailureCount += 0;
-    }
+//on having this function we are able to reflect the changes on transmissionFailureCount in both the files, or else the transmissionFailureCount remains 0 on increment
+function setTransmissionFailureCount(){

```

Comment:
Convention is that when a function returns some value, we start its name with a `get...`
A `set...` normally means this function set the value of the variable.